hypothesis_sqlalchemy/columnar/records.py

```diff
@@ -46,8 +46,11 @@ def lists_factory(columns: List[Column],
                       if is_column_unique(column)]
 
     if unique_indices:
-        def unique_by(row: Tuple[Any, ...]) -> Tuple[Any, ...]:
-            return tuple(row[index] for index in unique_indices)
+        # Create a tuple of functions, each function asserting the uniqueness
+        # of a single column value
+        unique_by = tuple(
+            (lambda row, idx=idx: row[idx]) for idx in unique_indices
+        )
```

Comment:
can we also add support for [unique constaints](https://docs.sqlalchemy.org/en/13/core/constraints.html#unique-constraint)?

Reply:
it looks like a complex separate task (because we should add those to tables strategy), so I can open a separate issue though

Reply:
created #17, will do in a spare time